stixcore/io/soc/manager.py

```diff
@@ -0,0 +1,89 @@
+import os
+from pathlib import Path
+from binascii import unhexlify
+from xml.etree import ElementTree as Et
+from collections import defaultdict
+
+from stixcore.tmtc.packets import TMTC, TMPacket
+from stixcore.util.logging import get_logger
+
+__all__ = ['SOCManager', 'SOCPacketFile']
+
+
+logger = get_logger(__name__)
+
+
+class SOCPacketFile:
+    def __init__(self, file):
+        self.file = Path(file)
+        if not self.file.exists():
+            raise ValueError(f"file path not found: {file}")
+        stat = self.file.stat()
+        self.size = stat.st_size
+        self.file_date = stat.st_ctime
+        self.spacecraft_date = None  # TODO some file name conventions?
+        self.tmtc = TMTC.All
+        self.packet_data = defaultdict(list)
+
+        with open(self.file, "r") as f:
+            header = f.read(1000)
+
+            if 'Response' not in header:
+                raise ValueError("xml does not contain any 'Response'")
+            if 'PktRawResponse' in header:
+                self.tmtc = TMTC.TM
+            elif 'PktTcReportResponse' in header:
+                self.tmtc = TMTC.TC
+            else:
+                raise ValueError("xml does not contain any TM or TC response")
+
+    def free_packet_data(self):
+        self.packet_data = defaultdict(list)
+
+    def read_packet_data_header(self):
+        root = Et.parse(str(self.file)).getroot()
+        if self.tmtc == TMTC.TC:
+            for i, node in enumerate(root.iter('PktTcReportListElement')):
+                # TODO add TC packer reading
+                pass
+        elif self.tmtc == TMTC.TM:
+            for i, node in enumerate(root.iter('Packet')):
+                packet_binary = unhexlify(node.text)
+                # Not sure why guess and extra moc header
+                tm_packet = TMPacket(packet_binary[76:])
```

Comment:
I'm wondering if the actual Parsing doesn't belong in the `processing/leveb/LevelB` class - better separation of concerns?

Reply:
moved